core/src/main/java/com/netease/arctic/scan/TableEntriesScan.java

```diff
@@ -309,11 +310,13 @@ private DataFile buildDataFile(StructLike fileRecord) {
     String filePath = fileRecord.get(dataFileFieldIndex(DataFile.FILE_PATH.name()), String.class);
     Long fileSize = fileRecord.get(dataFileFieldIndex(DataFile.FILE_SIZE.name()), Long.class);
     Long recordCount = fileRecord.get(dataFileFieldIndex(DataFile.RECORD_COUNT.name()), Long.class);
+    FileFormat format = FileFormat.fromFileName(filePath);
     DataFiles.Builder builder =
         DataFiles.builder(table.spec())
             .withPath(filePath)
             .withFileSizeInBytes(fileSize)
-            .withRecordCount(recordCount);
+            .withRecordCount(recordCount)
+            .withFormat(format != null ? format : FileFormat.PARQUET);
```

Comment:
We cannot assume the default table format is `PARQUET`, maybe we can find the file format in the record with filed named `file_format`.

Reply:
Make sense to me.